### PR TITLE
Refactor Watcher to return Connection object

### PR DESCRIPTION
**Pull request by mthenw**
@sebito91 Here are the changes that I've mentioned in your PR.

### CODE REVIEW

File: cmd/connector/main.go [hunk 1]
```diff
@@ -36,18 +39,33 @@ func main() {
...
 
 	// Server
-	logger.Debugf("Starting Config API on port: 4002")
-	logger.Fatal(httpapi.StartConfigAPI(store))
+	srv := httpapi.ConfigAPI(store)
+	go func() {
+		logger.Debugf("Starting Config API on port: 4002")
+		logger.Fatal(srv.ListenAndServe())
+	}()
+	defer srv.Shutdown(context.TODO())
```
Comment (sebito91) [inline cmd/connector/main.go:63]: Love this! :heart:

File: watcher/watcher.go [hunk 2]
```diff
@@ -2,51 +2,78 @@ package watcher
...
-				if watchEvent.Kv.Value == nil {
-					event.Type = EventDeleted
+				event := &Event{ID: extractIDFromKey(watchEvent.Kv.Key)}
+
+				if watchEvent.Type == mvccpb.PUT {
+					event.Type = Created
+
+					conn := &connection.Connection{}
+					if err := json.Unmarshal(watchEvent.Kv.Value, conn); err != nil {
+						w.log.Errorf("unmarshaling payload failed: %s", err)
```
Comment (sebito91) [inline watcher/watcher.go:71]: Add a `continue` here otherwise we send empty struct down chan?

**Review by sebito91 — CHANGES_REQUESTED** [review]
Small nit, otherwise LGTM. Good call on the context handler for the API.

File: watcher/watcher.go (hunk 2 above)
Comment (mthenw) [reply watcher/watcher.go:71]: true

**Review by sebito91 — APPROVED** [review]
LGTM!

**Pull request merged**